rwengine/src/ai/CharacterControllerActivities.hpp

```diff
@@ -0,0 +1,82 @@
+#pragma once
+
+#ifndef _CONTROLLER_ACTIVITY_HPP_
+#define _CONTROLLER_ACTIVITY_HPP_
+
+#include <ai/CharacterController.hpp>
+
+
+#define DECL_ACTIVITY(act_name)                                               \
+	class act_name : public CharacterController::Activity                 \
+	{                                                                     \
+	public:                                                               \
+		static constexpr auto ActivityName = #act_name;               \
+		std::string name() const override { return ActivityName; };   \
+		bool update(CharacterObject*, CharacterController*) override;
+
+#define DECL_ACTIVITY_CANSKIP(act_name)                          \
+	DECL_ACTIVITY(act_name)                                  \
+		bool canSkip(CharacterObject *) const override;
```

Comment:
Urgh, I think this is really ugly, but it's fine for this PR.

Generally: I don't even like that we had the `DECL_ACTIVITY` macro to begin with.
This extends on the idea of using macros; which is something we shouldn't be doing in a modern C++ project.

Are there any alternatives to this? I guess this could be implemented using some kind of template?
(for the future, it's beyond the scope of this PR)